source/common/router/scoped_rds.cc

```diff
@@ -330,17 +346,24 @@ void ScopedRdsConfigSubscription::onConfigUpdate(
           fmt::format("scope key conflict found, first scope is '{}', second scope is '{}'",
                       scope_name_by_key_hash[key_fingerprint], scope_name));
     }
+    // if new scope have key conflict with scope that is not going to be removed
+    if (scope_name_by_hash_.find(key_fingerprint) != scope_name_by_hash_.end() &&
+        scope_name_by_hash_[key_fingerprint] != scope_name &&
+        removed_scopes.find(scope_name_by_hash_[key_fingerprint]) == removed_scopes.end()) {
```

Comment:
for delta API, lets say previously it's {scope1:h1}, the update is ({scope2:h1, scope1:h2},removed={})
will the update gets rejected here?

Reply:
yes, it will gets rejected because scope2:h1 has key conflict with old scope with different name scope1:h1. This can be fixed easily though. Just marked all updated scopes as removed scope.

Reply:
Maybe I should add a unit test to cover this corner case?